iree/task/task.c

```diff
@@ -287,13 +287,8 @@ static uint32_t iree_math_ptr_to_xrgb(const uintptr_t ptr) {
   // This is just a simple hack to give us a unique(ish) per-pointer color.
   // It's only to make it easier to distinguish which tiles are from the same
   // dispatch.
-#if defined(IREE_PTR_SIZE_32)
-  return (uint32_t)ptr;
-#elif defined(IREE_PTR_SIZE_64)
-  return (uint32_t)ptr ^ (uint32_t)(ptr >> 32);
-#else
-#error Unknown pointer size
-#endif
+  uint64_t ptr64 = ptr;
+  return (uint32_t)ptr64 ^ (uint32_t)(ptr64 >> 32);
```

Comment:
Revert of https://github.com/google/iree/pull/5490 / replacement with `uint64_t` version intended?

Reply:
Yes #5490 didn't seem to fix the problem. It was still complaining about shifting by too much, which means that whatever this macro was detecting, it wasn't working as intended here, so we needed to do something like this anyway, at which point I suggested just dropping the macro, since this should work on any system that supports uint64_t (not technically required by the standard, but already assumed throughout the IREE runtime)